src/ConstructiveSolidGeometry/LinePrimitives/Line.jl

```diff
@@ -1,8 +1,29 @@
-struct Line{T} <: AbstractLinePrimitive{T}
+struct Line{T<:AbstractFloat} <: AbstractLinePrimitive{T}
     origin::CartesianPoint{T}
     direction::CartesianVector{T}
 end
 
+#Type promotion happens here
+function Line(origin::PT, direction::DIR) where {PT<:(CartesianPoint), DIR<:(CartesianVector)}
```

Comment:
I don't think we need brackets here around the types`PT <: (CartesianPoint)` -> `PT <: CartesianPoint, ...`